hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1950,3 +1966,26 @@ func setExpirationTimestampOnToken(ctx context.Context, c client.Client, tokenSe
 	tokenSecret.Annotations[hyperv1.IgnitionServerTokenExpirationTimestampAnnotation] = time.Now().Add(timeUntilExpiry).Format(time.RFC3339)
 	return c.Update(ctx, tokenSecret)
 }
+
+func (r *NodePoolReconciler) doesCPODecompressAndDecodeConfig(ctx context.Context, hcluster *hyperv1.HostedCluster) (bool, error) {
```

Comment:
Ah ok, so now during the upgrade, while the old ignition server (CPO) is active, this will return false, until some future sync sees the upgraded CPO with the label you added in the Dockerfile, which will from that point onwards encode and compress, right?